### PR TITLE
[IMP] pos_restaurant: swap payment and order button

### PR DESCRIPTION
**Pull request by davidmonnom**
*: point_of_sale,pos_loyalty,pos_restaurant

Previously, the payment button was larger than the others and next to
the numpad. The other buttons were managed by a mixin and arranged
in a predefined order.

Now, the payment button only stays in place when the "order" button is
not present, currently it can only be present if the restaurant module
is installed and a printer configured.

When it is present, it takes the place of the payment button and the
latter is added to the mixins with the other buttons

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/pos_loyalty/static/src/js/Loyalty.js [hunk 1]
```diff
@@ -166,7 +166,8 @@ patch(PosGlobalState.prototype, "pos_loyalty.PosGlobalState", {
                 dbCoupon.points
             );
             this.couponCache[coupon.id] = coupon;
-            this.partnerId2CouponIds[coupon.partner_id] = this.partnerId2CouponIds[coupon.partner_id] || new Set();
+            this.partnerId2CouponIds[coupon.partner_id] =
+                this.partnerId2CouponIds[coupon.partner_id] || new Set();
```
Comment (rhe-odoo) [inline addons/pos_loyalty/static/src/js/Loyalty.js:170]: useless diff

File: addons/pos_loyalty/static/src/js/Loyalty.js [hunk 2]
```diff
@@ -196,7 +197,9 @@ patch(PosGlobalState.prototype, "pos_loyalty.PosGlobalState", {
     getLoyaltyCards(partner) {
         const loyaltyCards = [];
         if (this.partnerId2CouponIds[partner.id]) {
-            this.partnerId2CouponIds[partner.id].forEach(couponId => loyaltyCards.push(this.couponCache[couponId]));
+            this.partnerId2CouponIds[partner.id].forEach((couponId) =>
+                loyaltyCards.push(this.couponCache[couponId])
+            );
```
Comment (rhe-odoo) [inline addons/pos_loyalty/static/src/js/Loyalty.js:202]: useless diff

File: addons/pos_loyalty/static/src/js/Loyalty.js [hunk 3]
```diff
@@ -214,7 +217,8 @@ patch(PosGlobalState.prototype, "pos_loyalty.PosGlobalState", {
                     partner.id,
                     points
                 );
-                this.partnerId2CouponIds[partner.id] = this.partnerId2CouponIds[partner.id] || new Set();
+                this.partnerId2CouponIds[partner.id] =
+                    this.partnerId2CouponIds[partner.id] || new Set();
```
Comment (rhe-odoo) [inline addons/pos_loyalty/static/src/js/Loyalty.js:221]: useless diff

File: addons/point_of_sale/static/src/js/Screens/ProductScreen/ActionpadWidget.js [hunk 4]
```diff
@@ -19,4 +19,16 @@ export class ActionpadWidget extends Component {
...
+        const order = this.pos.globalState.get_order();
+
+        if (order.orderlines.length) {
+            order.pay();
+        }
+    }
+
+    get highlightPay() {
+        return this.pos.globalState.get_order().orderlines.length;
+    }
```
Comment (rhe-odoo) [inline addons/point_of_sale/static/src/js/Screens/ProductScreen/ActionpadWidget.js:33]: is it used?

File: addons/pos_restaurant/static/src/js/Screens/ProductScreen/ControlButtons/SubmitOrderButton.js [hunk 5]
```diff
@@ -17,24 +16,14 @@ export class SubmitOrderButton extends Component {
...
-                    } else {
-                        this.popup.add(ErrorPopup, {
-                            title: this.env._t("Printing failed"),
-                            body: this.env._t("Failed in printing the changes in the order"),
-                        });
-                    }
-                }
+                this.pos.globalState.get_order().submitOrder();
             } finally {
                 this.clicked = false;
```
Comment (rhe-odoo) [inline addons/pos_restaurant/static/src/js/Screens/ProductScreen/ControlButtons/SubmitOrderButton.js:28]: this clicked will never be true again. Finally close might be removed

**Review by trj-odoo — CHANGES_REQUESTED** [review]

File: addons/point_of_sale/static/src/js/models.js [hunk 6]
```diff
@@ -2923,6 +2924,28 @@ export class Order extends PosModel {
...
+                confirmText: _t("Yes"),
+                cancelText: _t("No"),
+            });
+            if (confirmed) {
+                this.pos.env.services.pos.showScreen("PaymentScreen");
+            }
+        } else {
+            this.pos.env.services.pos.showScreen("PaymentScreen");
+        }
+    }
```
[the remark this answers is not in the document]
Comment (sdegueldre) [reply addons/point_of_sale/static/src/js/models.js:2948]: Seems fine, the end goal is for models to have direct access to the pos service so this doesn't need to be so verbose

File: addons/pos_restaurant/data/pos_restaurant_demo.xml [hunk 7]
```diff
@@ -200,7 +200,6 @@
             <field name="iface_splitbill">True</field>
             <field name="iface_printbill">True</field>
             <field name="iface_orderline_notes">True</field>
-            <field name="printer_ids" eval="[(6, 0, [ref('pos_restaurant.kitchen_printer')])]" />
```
[the remark this answers is not in the document]
Comment (davidmonnom) [reply addons/pos_restaurant/data/pos_restaurant_demo.xml:203]: This line added the printer to the pos_config without enabling the printer option. This caused a network error when trying to send the command. This was a misconfiguration of the data demos

**Review by trj-odoo — APPROVED** [review]

**Pull request closed without merging**